splinepy/helpme/create.py

```diff
@@ -277,6 +287,143 @@ def from_bounds(parametric_bounds, physical_bounds):
     return bspline_box
 
 
+def determinant_spline(spline):
+    """
+    Creates a spline representing the projection of the jacobian determinant
+
+    Only works if (parameter dimension == physical dimension) and non-rational splines
+    or rational splines with equal weights
+    (otherwise give B-Spline approximation of spline)!
+
+    dmin > 0 --> spline is not tangled
+    dmin <= 0 --> spline is COULD BE tangled
+
+    references:
+    Mantzaflaris, A., Jüttler, B., Khoromskij, B. N., & Langer, U. (2017)
+    Limkilde, A., Evgrafov, A., Gravesen, J., & Mantzaflaris, A. (2021)
+
+    Parameters
+    ----------
+    None
+
+    Returns
+    -------
+    determinant_projection: BSpline
+      Spline which represents the jacobian determinant of the given spline object.
+    """
+
+    # checks if dimensions are equal.
+    if spline.dim != spline.para_dim:
+        raise ValueError(
+            f"Dimension of parameter space and physical space ({spline.whatami})"
+            "are not the same!"
+        )
+
+    # checks if spline is rational and if weights differ
+    # otherwise NURBS can be treated like regular BSpline
+
+    if spline.is_rational and np.allclose(
+        spline.weights, spline.weights.ravel()[0]
+    ):
+        rational_spline = True
```

Comment:
Even if the weights are non-zero, the determinant projection should give the best possible approximation. So don't modify the weights, keep them as they are and just warn the user about the potential errors

Reply:
Okay.